lib/awful/ewmh.lua

```diff
@@ -270,6 +275,10 @@ function ewmh.geometry(c, context, hints)
             props.honor_workarea = honor_default
         end
 
+        if props.honor_padding == nil and props.honor_workarea then
+            props.honor_padding = beautiful.maximized_honor_padding ~= false
+        end
```

Comment:
The commit message says that this is specific to maximisation, but I fail to see that in the code...?